tools/swift-machine-learning-api/README.md

```diff
@@ -0,0 +1,336 @@
+
+The trained model are saved into different versions based on the name you give to these versions.Version control is important as there will be frequent data changes or hyperparameters updates, which will then create a different model.
+
+## Functionalities of the API
+
+All the below functionalities can be used with a simple post request to the respective APIs with very basic background knowledge.
+ - To train various ML models on GCP CMLE
+ - To deploy the trained model on GCP CMLE
+ - To predict results of the deployed model both on a batch or on a single datapoints using GCP CMLE 
+ - To visualize the predicted results using LIME functionality 
+
+## Structure of code
+```
+SMLA
+    |__ main.py  (entry point of the API)                                                
+    |__ train.py  (train functions)
+    |__ predict.py  (predict functions)        
+    |__ deploy.py  (deploy functions)
+    |__ lime_utils.py  (utils functions for LIME functionalities)
+    |__ update.sh  (bash script for updating trainer package) 
+    |__ config
+        |__ train.yaml  (yaml file for train configurations)
+        |__ config_file.yaml  (yaml file for API configurations)
+        |__ developer.yaml  (yaml file for developer configurations)
+    |__ codes                   
+        |__ config.yaml  (yaml file for CloudML configurations)
+        |__ setup.py 
+        |__ trainer  (trainer package)
+            |__ __init__.py
+            |__ launch_demo.py  (end-to-end script for running package)
+            |__ models.py  (tensorflow estimator functions)
+            |__ input_pipeline_dask.py  (handles input_pipeline)
+            |__ utils
+                |__ __init__.py
+                |__ custom_utils_fn.py
+                |__ optimizer_utils.py
+                |__ metric_utils.py
+```
+
+## Instructions to run the code
+
+### Configuring the API
+
+- Create a config file named **_'config_file.yaml'_** like below, with your project id, bucket name and path of service account key along for main.py in the **config directory** for interacting with GCP resources.
+
+  ```
+  bucket_name: "gs://<bucket-name>"
+  project: <project_name>
+  project_id: <project_ID>
+  service_account_json_key: <path/of/service/account/key/json/file>
+  ```
+
+- Create a config file named **_'train.yaml'_** like below along for train.py in the **config directory** to specify train configurations. Refer to [*CMLE machine types*](https://cloud.google.com/ml-engine/docs/tensorflow/machine-types) for specifying type virtual machines you want the model to be trained. The trainer package will be created and updated in GCS bucket by running bash script update.sh. 
```

Comment:
It seems this file already exists. Should the user edit the exisiting train.yaml in the config directory?

Reply:
Actually only developer.yaml is pre-prepared in the config/. The train.yaml and config_file.yaml are supposed to be created by the user.